lib/action_cable/connection/base.rb

```diff
@@ -180,6 +180,11 @@ def allow_request_origin?
           end
         end
 
+        def allowed_origins_match? origin
+          allowed_origins = Array(server.config.allowed_request_origins)
+          allowed_origins.any? { |allowed_origin| allowed_origin.is_a?(Regexp) ? allowed_origin =~ origin : allowed_origin == origin }
```

Comment:
You could use the `===` operator to match both strings and regexes, so the class check wouldn't be required.

```
allowed_origins.any? { |allowed_origin| allowed_origin === origin }
```